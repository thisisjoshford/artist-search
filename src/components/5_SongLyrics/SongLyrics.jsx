import React from 'react';
import PropTypes from 'prop-types';

const SongLyrics = ({ songName, songLyrics }) => (
  <>
    <h3>{songName}</h3>
    <p>{songLyrics} </p>
  </>
);

SongLyrics.propTypes = {
  songName: PropTypes.string.isRequired,
  songLyrics: PropTypes.array.isRequired
};

export default SongLyrics;
