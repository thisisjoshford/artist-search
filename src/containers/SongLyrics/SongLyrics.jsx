import React, { useState, useEffect } from 'react';

import { useParams } from 'react-router-dom';
import { getLyrics } from '../../services/apiFetches';

const SongLyricsControls = () => {

  const [songLyrics, setSongLyrics] = useState([]);
  const { artistName, songName } = useParams();

  useEffect(() => {
    getLyrics(artistName, songName)
      .then(res => setSongLyrics(res));
  }, []);

  console.log(songLyrics);

  return (
    <>
      <SongLyrics artistName={artistName} songLyrics={songLyrics}/>
    </>
  );
};

export default SongLyricsControls;
