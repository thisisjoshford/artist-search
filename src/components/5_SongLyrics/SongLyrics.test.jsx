import React from 'react';
import { shallow } from 'enzyme';
import SongLyrics from './SongLyrics';

describe('Search component', () => {
  it('renders Search', () => {
    const songLyrics = {
      lyrics: 'doo doo doo'
    };

    const wrapper = shallow(
      <SongLyrics artistName='Tool' songName='Song1' songLyrics='doo doo doo doo' SongLyrics={songLyrics}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
