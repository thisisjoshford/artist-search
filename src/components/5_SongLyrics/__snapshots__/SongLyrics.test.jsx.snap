// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component renders Search 1`] = `
<section>
  <h3>
    Lyrics for 
    Song1
     by 
    Tool
  </h3>
  <pre />
</section>
`;
